docs/application/native/guides/ui/tizen-ws-shell.md

```diff
@@ -0,0 +1,151 @@
+
+```
+#include <tzsh_quickpanel.h>
+```
+
+
+## Creating TZSH Quickpanel Handler
+Once you have created the main window for your application, call the `tzsh_quickpanel_create_with_type` function with a native window ID to create `tzsh_quickpanel_h` structure (in [mobile](../../api/mobile/latest/group__TIZEN__WS__SHELL__QUICKPANEL__MODULE.html#gaaa00e8e25b43c9538ca188bc43bdb3ac) and [wearable](../../api/wearable/latest/group__TIZEN__WS__SHELL__QUICKPANEL__MODULE.html#gaaa00e8e25b43c9538ca188bc43bdb3ac) applications).
+
+```
+#include <Elementary.h>
+
+static void
+init(const char *name)
+{
+   Evas_Object *main_win;
+
+   /* create main window for the application */
+   main_win = elm_win_util_standard_add(name, name);
+
+   /* set up main window */
+   ...
+   evas_object_show(main_win);
+
+
+   tzsh_h tzsh;
+   tzsh_quickpanel_h tzsh_qp;
+   tzsh_window tz_win;
+
+   /* Get native window ID of main window */
+   tz_win = elm_win_window_id_get(main_win);
+
+   /* Create tzsh_h structure */
+   tzsh = tzsh_create(TZSH_TOOLKIT_TYPE_EFL);
+
+   /* Create tzsh_quickpanel_h structure */
+   tzsh_qp = tzsh_quickpanel_create_with_type(tzsh, tz_win, TZSH_QUICKPANEL_TYPE_SYSTEM_DEFAULT);
+```
+
+
+## Showing and Hiding Quickpanel Window
+
+To show and hide quickpanel window when your application's window is activated, call the following functions:
+
+```
+   ...
+
+   if (show)
+     tzsh_quickpanel_show(tzsh_qp); /* To show quickpanel window */
+   else
+     tzsh_quickpanel_hide(tzsh_qp); /* To hide quickpanel window */
+
+   ...
+```
+
+## Getting Visible Status of Quickpanel Window
+To know the state of current visibility of qp service's window, call the `tzsh_quickpanel_visible_get` function. The quickpanel service's window may be visible or not visible on the screen.
+
+```
+   ...
+   tzsh_quickpanel_state_visible_e state;
+
+   tzsh_quickpanel_visible_get(tzsh_qp, &state);
+   if (state == TZSH_QUICKPANEL_STATE_VISIBLE_SHOWN)
+     /* visible state */
+     ;
+   else if (state == TZSH_QUICKPANEL_STATE_VISIBLE_HIDDEN)
+     /* invisible state */
+     ;
+   else
+     /* error */
+     ;
+```
+
+
+## Registering Changed Event for the Quickpanel Window
+To be notified of state changes, implement the appropriate event callback function and call the `tzsh_quickpanel_event_handler_add` function with that. If you want to change your application's behavior to match visibility of the quicknapel service's window, then you should handle state change event as following:
+
+```
+static tzsh_quickpanel_event_handler_h handler;
+
+static void
+ev_callback(int type, tzsh_quickpanel_event_info_h ev_info, void *data)
+{
+   tzsh_quickpanel_state_visible_e state;
+
+   if (type != TZSH_QUICKPANEL_EVENT_VISIBLE)
+     return;
+
+   tzsh_quickpanel_event_visible_get(ev_info, &state);
+   if (state == TZSH_QUICKPANEL_STATE_VISIBLE_SHOWN)
+     ;
```

Comment:
line 115, line 117 and line 119. How about replacing the ";"s with comments for each case, as like on line 86?

Reply:
The sample seems to me asking a user to write down his own code for each case of "quickpanel is visible/invisible/something-went-wrong".  If the functions for each case are empty, I think leaving comments for each case to tell a user what to do would be better than ";".  
"
   if (state == TZSH_QUICKPANEL_STATE_VISIBLE_SHOWN)
     ;
"
means "If the quickpanel is visible, ". 
It only checks the state and then just finish the line.

Reply:
Hello, @annie-samsung.
We have same issue with above here again.

Reply:
@gl-lee , update with
if (state == TZSH_QUICKPANEL_STATE_VISIBLE_SHOWN)
{
/* Do what you want with visible quickpanel*/
;
}

Reply:
@mijin85cho will wait to hear from the developer

Reply:
Same as above. I will change example code as EFL UI style without changing comments.